services/provider/server/server.go

```diff
@@ -690,3 +705,34 @@ func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStat
 
 	return &pb.ReportStatusResponse{}, nil
 }
+
+// PeerBlockPool RPC call to send the bootstrap secret for the pool
+func (s *OCSProviderServer) PeerBlockPool(ctx context.Context, req *pb.PeerBlockPoolRequest) (*pb.PeerBlockPoolResponse, error) {
+
+	if err := s.cephRBDMirrorManager.Create(ctx); err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to create RBDMirror instance: %v", err)
+	}
+
+	// enable mirroring on blockPool in the req
+	if err := s.cephBlockPoolManager.EnableBlockPoolMirroring(ctx, req.BlockPoolName); err != nil {
+		if kerrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "Failed to find CephBlockPool resource %s: %v", req.BlockPoolName, err)
```

Comment:
I believe the errors are a bit misleading (same for remaining), basically the function asks for enabling mirroring and if you didn't find the source why are we at this line 🤔?

by source I mean, block pool, at the start why can't we return if we didn't find the blockpool? basically we are checking blockpool existence indirectly three times throughtout this function.

Reply:
updated to return at the beginning of the function if the blockpool doesn't exist